File: episode-51-r/fizzbuzz2.r
#!/usr/bin/env Rscript

i = seq(1, 100)
x = i
x[i %% 3 == 0] = "Fizz"
x[i %% 5 == 0] = "Buzz"
x[i %% 15 == 0] = "FizzBuzz"
cat(x, sep="\n")

File: episode-100-ruby/fizzbuzz2.rb
#!/usr/bin/env ruby

def divisible_by(n)
  proc{|m| m % n == 0}
end

puts (1..100).map{|n|
  case n
  when divisible_by(15)
    "FizzBuzz"
  when divisible_by(5)
    "Buzz"
  when divisible_by(3)
    "Fizz"
  else
    n
  end
}
